public/app/core/components/RolePicker/RolePickerInput.tsx

```diff
@@ -47,8 +49,8 @@ export const RolePickerInput = ({
 
   return !isFocused ? (
     <div className={styles.selectedRoles} onMouseDown={onOpen}>
-      <ValueContainer>{builtInRole}</ValueContainer>
-      {!!numberOfRoles && (
+      {showBuiltInRole && <ValueContainer>{builtInRole}</ValueContainer>}
+      {!!numberOfRoles ? (
```

Comment:
nit: This is getting quite hard to read and reason about. Could this part be simplified somehow? Maybe extracting smaller components ?